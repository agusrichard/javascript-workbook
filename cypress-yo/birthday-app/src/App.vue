<template>
  <v-app>
    <v-main>
      <Layout>
        <router-view />
      </Layout>
    </v-main>
  </v-app>
</template>

<script>
import Layout from './components/layout'
export default {
  components: { Layout }
}
</script>
