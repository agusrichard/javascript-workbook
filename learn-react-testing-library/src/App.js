import { useState } from 'react'

import Example from './Example'

function App() {
  const [showMe, setShowMe] = useState(false)

  return (
    <div className="App">
      <div>Sekardayu Hana Pradiani</div>
      <button onClick={() => setShowMe(!showMe)}>Click Me!</button>
      { showMe && <div>Agus Richard Lubis</div> }
      <Example title='Its a Me. Malario' />
    </div>
  )
}

export default App
