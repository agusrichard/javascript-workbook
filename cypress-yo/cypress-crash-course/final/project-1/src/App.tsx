import logo from './logo.svg';
import './App.css';
import SideNav from './components/SideNav/SideNav';
import Habit from "./components/Habit/Habit"
import Accomplishment from './components/Accomplishment/Accomplishment';
import { BrowserRouter, Route, Switch } from "react-router-dom"
import Rewards from './components/Rewards/Rewards';
import Elements from './components/Elements/Elements';

function App() {
  return (
    <BrowserRouter>
      <div className="App">
        <div className="App-container">
          <SideNav />
          <Switch>
            <Route strict exact path="/habits" component={Habit} />
            <Route strict exact path="/accomplishments" component={Accomplishment} />
            {/* <Route strict exact path="/rewards" component={Rewards} /> */}
            <Route strict exact path="/elements" component={Elements} />
          </Switch>
        </div>
      </div>
    </BrowserRouter>
  );
}

export default App;
