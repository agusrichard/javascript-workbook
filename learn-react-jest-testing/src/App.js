import './App.css';
import Account from './Account'

function App() {
  const user = {
    name: 'Sekardayu',
    age: 23,
    email: 'sekardayu@example.com'
  }
  
  return (
    <div className="App">
      <h1>Sekardayu Hana Pradiani</h1>
      <h1>Saskia Nurul Azhima</h1>
      <Account user={user} />
    </div>
  );
}

export default App;
