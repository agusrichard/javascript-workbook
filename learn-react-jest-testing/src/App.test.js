import React from 'react'
import { shallow } from 'enzyme'

import App from './App'
import toJson from 'enzyme-to-json'

it('App contains h1', () => {
  const wrapper = shallow(<App />)
  const header = <h1>Sekardayu Hana Pradiani</h1>

  expect(wrapper.contains(header)).toEqual(true)
})

it('Snapshot test', () => {
  const tree = shallow(<App />)
  expect(toJson(tree)).toMatchSnapshot()
})