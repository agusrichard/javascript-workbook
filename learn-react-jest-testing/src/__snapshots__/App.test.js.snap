// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div
  className="App"
>
  <h1>
    Sekardayu Hana Pradiani
  </h1>
  <h1>
    Saskia Nurul Azhima
  </h1>
  <Account
    user={
      Object {
        "age": 23,
        "email": "sekardayu@example.com",
        "name": "Sekardayu",
      }
    }
  />
</div>
`;
